crash when changing case of vault_identity_group name

### Terraform Version
```
Terraform v1.1.5
on darwin_arm64
+ provider registry.terraform.io/hashicorp/aws v4.1.0
+ provider registry.terraform.io/hashicorp/azuread v2.5.0
+ provider registry.terraform.io/hashicorp/null v3.1.0
+ provider registry.terraform.io/hashicorp/random v3.1.0
+ provider registry.terraform.io/hashicorp/vault v3.2.1
+ provider registry.terraform.io/scottwinkler/shell v1.7.10
```

### Affected Resource(s)
Please list the resources as a list, for example:
- vault_identity_group
- vault_identity_group_alias

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files
```hcl
locals {
  vault_assignments = {
    engineering = [
    #Engineering = [
      "default",
    ]
  }
}

data "azuread_groups" "azure" {
   display_name = keys(local.vault_assignments)
}

resource "vault_identity_group" "azure" {
  for_each = local.vault_assignments
  name     = each.key
  type     = "external"
  policies = each.value
 }
resource "vault_identity_group_alias" "azure" {
  for_each       = local.vault_assignments
  name           = data.azruead_groups[each.key].id
  mount_accessor = vault_jwt_auth_backend.azure.accessor
  canonical_id   = vault_identity_group.azure[each.key].id
}
```

### Debug Output
No debugging output was generated.

### Panic Output

```
Stack trace from the terraform-provider-vault_v3.2.1_x4 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x30 pc=0x1059dbc54]

goroutine 132 [running]:
github.com/hashicorp/terraform-provider-vault/vault.identityGroupCreate(0x14000adf000, {0x105ec7b60, 0x140002dba40})
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/hashicorp/terraform-provider-vault/vault/resource_identity_group.go:189 +0x434
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x1400054b6c0, {0x105f06be8, 0x14000595600}, 0x14000adf000, {0x105ec7b60, 0x140002dba40})
        /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.0/helper/schema/resource.go:329 +0x170
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0x1400054b6c0, {0x105f06be8, 0x14000595600}, 0x140005e1930, 0x14000adee80, {0x105ec7b60, 0x140002dba40})
        /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.0/helper/schema/resource.go:467 +0x8d8
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0x14000284528, {0x105f06be8, 0x14000595600}, 0x14000aaa8c0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-sdk/v2@v2.10.0/helper/schema/grpc_provider.go:977 +0xe38
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0x1400013aa80, {0x105f06c90, 0x140005edcb0}, 0x140001a2ee0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/tf5server/server.go:603 +0x338
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x105e915c0, 0x1400013aa80}, {0x105f06c90, 0x140005edcb0}, 0x1400088d0e0, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/github.com/hashicorp/terraform-plugin-go@v0.5.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x1c0
google.golang.org/grpc.(*Server).processUnaryRPC(0x140005048c0, {0x105f1b580, 0x1400075c000}, 0x14000a605a0, 0x140004c7c50, 0x1066a0da0, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.41.0/server.go:1279 +0xc60
google.golang.org/grpc.(*Server).handleStream(0x140005048c0, {0x105f1b580, 0x1400075c000}, 0x14000a605a0, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.41.0/server.go:1608 +0xa34
google.golang.org/grpc.(*Server).serveStreams.func1.2(0x1400003ee40, 0x140005048c0, {0x105f1b580, 0x1400075c000}, 0x14000a605a0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.41.0/server.go:923 +0x94
created by google.golang.org/grpc.(*Server).serveStreams.func1
        /opt/teamcity-agent/work/5d79fe75d4460a2f/pkg/mod/google.golang.org/grpc@v1.41.0/server.go:921 +0x1f0

Error: The terraform-provider-vault_v3.2.1_x4 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Expected Behavior

Ideally, handled gracefully or rejected the change because of duplicate values (I know Vault lower-cases things internally).

### Actual Behavior
Crashed

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`
2. change the case of the identity group name
2. `terraform apply`

### Important Factoids
None

### References
None

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,3 +1,7 @@
+## 3.4.0 (Unreleased)
+BUGS:
+* `resource/identity_group`: Report an error upon duplicate resource creation failure. Document group name caveats. ([#1352](https://github.com/hashicorp/terraform-provider-vault/pull/1352))
+
 ## 3.3.0 (February 17, 2022)
 FEATURES:
 * Add KMIP support: new resources `vault_kmip_secret_backend`, `vault_kmip_secret_scope` and `vault_kmip_secret_role` ([#1339](https://github.com/hashicorp/terraform-provider-vault/pull/1339))

--- a/vault/resource_identity_group.go
+++ b/vault/resource_identity_group.go
@@ -1,6 +1,7 @@
 package vault
 
 import (
+	"errors"
 	"fmt"
 	"log"
 
@@ -183,17 +184,18 @@ func identityGroupCreate(d *schema.ResourceData, meta interface{}) error {
 
 	if resp == nil {
 		path := identityGroupNamePath(name)
-		groupMsg := "Unable to determine group id."
-
-		if group, err := client.Logical().Read(path); err == nil {
-			groupMsg = fmt.Sprintf("Group resource ID %q may be imported.", group.Data["id"])
+		resp, err := client.Logical().Read(path)
+		if err == nil {
+			err = errors.New("unknown")
+			if resp != nil {
+				err = fmt.Errorf(
+					"group already exists with path=%q, id=%q", path, resp.Data["id"])
+			}
 		}
-
-		return fmt.Errorf("Identity Group %q already exists. %s", name, groupMsg)
-	} else {
-		log.Printf("[DEBUG] Wrote IdentityGroup %q", resp.Data["name"])
+		return fmt.Errorf("failed to create identity group %q, reason=%w", name, err)
 	}
 
+	log.Printf("[DEBUG] Created IdentityGroup %q", resp.Data["name"])
 	d.SetId(resp.Data["id"].(string))
 
 	return identityGroupRead(d, meta)

--- a/vault/resource_identity_group_test.go
+++ b/vault/resource_identity_group_test.go
@@ -3,6 +3,7 @@ package vault
 import (
 	"encoding/json"
 	"fmt"
+	"regexp"
 	"strconv"
 	"strings"
 	"testing"
@@ -109,6 +110,39 @@ func TestAccIdentityGroupExternal(t *testing.T) {
 			{
 				Config: testAccIdentityGroupConfig(group),
 				Check:  testAccIdentityGroupCheckAttrs(),
+			},
+		},
+	})
+}
+
+func TestAccIdentityGroup_DuplicateCreate(t *testing.T) {
+	// group identity names are stored in lower case,
+	// this test attempts to create two resources with different casing for the
+	// same lower case group name.
+	group := fmt.Sprintf("test_group_%d", acctest.RandInt())
+	config := fmt.Sprintf(`
+resource "vault_identity_group" "test_lower" {
+  name     = %q
+  type     = "external"
+  policies = ["default"]
+}
+
+resource "vault_identity_group" "test_upper" {
+  name     = %q
+  type     = "external"
+  policies = ["default"]
+}
+`, group, strings.ToUpper(group[0:1])+group[1:])
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testutil.TestAccPreCheck(t) },
+		Providers:    testProviders,
+		CheckDestroy: testAccCheckIdentityGroupDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				ExpectError: regexp.MustCompile(
+					fmt.Sprintf(`(?i)failed to create identity group %q, reason=group already exists .+`, group)),
 			},
 		},
 	})

--- a/website/docs/r/identity_group.html.md
+++ b/website/docs/r/identity_group.html.md
@@ -42,6 +42,37 @@ resource "vault_identity_group" "group" {
 }
 ```
 
+## Caveats
+
+It's important to note that Vault identity groups names are *case-insensitive*. For example the following resources would be equivalent.
+Applying this configuration would result in the provider failing to create one of the identity groups, since the resources share the same `name`.
+
+This sort of pattern should be avoided:
+```hcl
+resource "vault_identity_group" "internal" {
+  # this duplicates the resource below
+  name     = "internal"
+  type     = "internal"
+  policies = ["dev", "test"]
+
+  metadata = {
+    version = "2"
+  }
+}
+
+resource "vault_identity_group" "Internal" {
+  # this duplicates the resource above
+  name     = "Internal"
+  type     = "internal"
+  policies = ["dev", "test"]
+
+  metadata = {
+    version = "2"
+  }
+}
+```
+
+
 ## Argument Reference
 
 The following arguments are supported:

